adding new Upgrade section to docs site (#5085)
zowe/docs-site: docs/upgrade/upgrade-zos.md

Author: ```suggestion
The `default.yaml` file is not meant to be edited, but you can observe changes and apply overrides within your Zowe YAML as desired.
```

Author: ```suggestion
Zowe also has an example file, `example-zowe.yaml`, which is often used to create your Zowe YAML during first-time installations. This file also changes from version to version, but is not used by Zowe, so if you do not take the changes from within it, you may miss out on new features and behaviors.
```

Author: ```suggestion
It is recommended to always review these files to check for changes that you may wish to apply to your own Zowe YAML.
```

Author: ```suggestion
It's recommended to always review these files to check for changes that you might wish to apply to your own Zowe YAML.
```

Author: ```suggestion
One way to review the changes is to use GitHub's compare tool.
```

# Upgrading
<!--This topic is being deprecated to make the parent topic more straightforward. See upgrading-zowe.md in the upgrade folder. -->

Performing an upgrade of Zowe involves following a subset of the first-time installation steps documented within the [Zowe z/OS components installation checklist](../user-guide/zos-components-installation-checklist.md), and updating your Zowe YAML. Exceptions are noted within this document.

:::warning

> Zowe cannot be upgraded while Zowe services are active. Stop all running Zowe instances before proceeding with the upgrade process. For detailed instructions, see [Stop Zowe](../user-guide/start-zowe-zos.md).

Reviewer: There is an article on how to update Zowe in HA without downtime - https://docs.zowe.org/stable/user-guide/api-mediation/upgrade-zowe-no-downtime/

Reviewer 2: Ok. This upgrade with HA article has not been in the sidebar, but we can certainly add it to this Upgrade section if it should be widely accessible.

Reviewer 2: revised

## Installation and Configuration

You should perform the steps in the "Installing" section, but you only need to complete some of the steps in the "Configuration" sections.

> Unless otherwise noted by an announcement or SMPE HOLD statement, you do not need to update certificates, instance datasets, networking, or security rules during an upgrade. Using [Configuring Zowe via JCL](../user-guide/configuring-zowe-via-jcl.md) as an example, you can skip the "Create Instance Datasets", "Grant SAF premissions", and "Keyring Tasks" steps. Therefore, the list of JCL configuration steps for upgrades becomes:

Author: ```suggestion
Unless otherwise noted by an announcement or SMPE HOLD statement, you do not need to update certificates, instance data sets, networking, or security rules during an upgrade. Using [Configuring Zowe via JCL](../user-guide/configuring-zowe-via-jcl.md) as an example, you can skip the "Create Instance Data sets", "Grant SAF permissions", and "Keyring Tasks" steps. Therefore, the list of JCL configuration steps for upgrades becomes:
```
